assembly/src/ast/module.rs

```diff
@@ -365,6 +369,32 @@ impl Module {
         self.procedures.iter_mut()
     }
 
+    /// Returns procedures exported from this module.
+    ///
+    /// Each exported procedure is represented by its global procedure index and a fully qualified
+    /// name. The global procedure index is computed using the specified module index.
+    pub fn exported_procedures(
+        &self,
+        module_idx: ModuleIndex,
```

Comment:
It feels a bit weird passing in `ModuleIndex` here. If you removed this parameter, and returned a `ProcedureIndex` rather than `GlobalProcedureIndex`, then it becomes a more generally useful function (as you no longer need to have a `ModuleIndex` to call it), _plus_ you can easily obtain the `GlobalProcedureIndex` in the caller if you have the `ModuleIndex` on hand.

More generally though, `ModuleIndex` is an internal implementation detail of the `Assembler`/`ModuleGraph`, so I would avoid leaking it into the AST where possible.

Reply:
Agreed. Refactored this to remove `module_idx` parameter.